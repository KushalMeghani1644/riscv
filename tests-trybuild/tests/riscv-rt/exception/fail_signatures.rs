#[riscv_rt::exception(riscv::interrupt::Exception::LoadMisaligned)]
fn my_exception(code: usize) {}

#[riscv_rt::exception(riscv::interrupt::Exception::StoreMisaligned)]
fn my_other_exception(trap_frame: &riscv_rt::TrapFrame, code: usize) {}

#[riscv_rt::exception(riscv::interrupt::Exception::LoadFault)]
async fn my_async_exception(trap_frame: &riscv_rt::TrapFrame, code: usize) {}

fn main() {}
