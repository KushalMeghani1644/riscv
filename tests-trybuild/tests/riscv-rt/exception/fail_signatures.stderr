error: `#[exception]` function must have signature `[unsafe] fn([&[mut] riscv_rt::TrapFrame]) [-> !]`
 --> tests/riscv-rt/exception/fail_signatures.rs:2:1
  |
2 | fn my_exception(code: usize) {}
  | ^^

error: `#[exception]` function must have signature `[unsafe] fn([&[mut] riscv_rt::TrapFrame]) [-> !]`
 --> tests/riscv-rt/exception/fail_signatures.rs:5:1
  |
5 | fn my_other_exception(trap_frame: &riscv_rt::TrapFrame, code: usize) {}
  | ^^

error: `#[exception]` function must have signature `[unsafe] fn([&[mut] riscv_rt::TrapFrame]) [-> !]`
 --> tests/riscv-rt/exception/fail_signatures.rs:8:1
  |
8 | async fn my_async_exception(trap_frame: &riscv_rt::TrapFrame, code: usize) {}
  | ^^^^^
